### ISSUE TITLE
System.IndexOutOfRangeException: Index was outside the bounds of the array

### ISSUE THREAD

**Issue by pluma9** [PowerShell/PSReadLine#2348]
### Environment
PSReadLine: 2.2.0-beta1
PowerShell: 7.1.2
OS: Microsoft Windows 10.0.18363
BufferWidth: 210
BufferHeight: 52

Last 200 Keys
```
 r e s t o r e Spacebar * Enter $ e v e n t f i l Tab [ 1 ] Enter Escape k f 1 2 Enter Escape k w w r 2 Enter Escape k w w r 3 Enter Escape k k k k k k k Escape w Escape w w w w b l l h d e i Backspace ỉ Escape x i $ e n v Backspace Backspace v e n Tab Tab Backspace f Tab [ 3 ] Escape Enter g i t Spacebar d i f f Enter Escape d d i g Spacebar Escape d d i g i t Spacebar r e s o t Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace g i t Spacebar r e s t o r e Spacebar * Enter Escape k k k k j Escape w f [ x x x 0 w w w w w w j j j j j j k k k k k k w Escape Escape j j j j k k k Enter g i t Spacebar d i f f Enter n v i m Spacebar ( f z f ) Enter Escape %
```

### Exception
```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at System.Text.StringBuilder.get_Chars(Int32 index)
   at Microsoft.PowerShell.PSConsoleReadLine.ViFindBrace(Int32 i)
   at Microsoft.PowerShell.PSConsoleReadLine.ViGotoBrace(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(PSKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken cancellationToken)
```

**Comment by daxian-dbw** [comment]
A simple repro:

1. Run `Set-PSReadLineOption -EditMode Vi`
2. Type `Escape`, `%`

The exception will then be triggered.

**Comment by ghost** [comment]
:tada: This issue was addressed in 2879, which has now been successfully released in [`v2.2.0-beta4`](https://github.com/PowerShell/PSReadLine/releases/tag/v2.2.0-beta4). :tada:
